test/jellyfish_web/controllers/component_controller_test.exs

```diff
@@ -321,6 +336,59 @@ defmodule JellyfishWeb.ComponentControllerTest do
     end
   end
 
+  describe "Create File Component" do
+    test "renders component with required options", %{conn: conn, room_id: room_id} do
+      conn =
+        post(conn, ~p"/room/#{room_id}/component",
+          type: "file",
+          options: %{filePath: @file_component_source}
+        )
+
+      assert response =
+               %{
+                 "data" => %{
+                   "id" => id,
+                   "type" => "file",
+                   "properties" => %{}
+                 }
+               } =
+               json_response(conn, :created)
+
+      assert_response_schema(response, "ComponentDetailsResponse", @schema)
+
+      conn = get(conn, ~p"/room/#{room_id}")
+
+      assert %{
+               "id" => ^room_id,
+               "components" => [
+                 %{"id" => ^id, "type" => "file"}
+               ]
+             } = json_response(conn, :ok)["data"]
+    end
+
+    test "renders error when component requires options not present in request", %{
+      conn: conn,
+      room_id: room_id
+    } do
+      conn = post(conn, ~p"/room/#{room_id}/component", type: "file")
+
+      assert json_response(conn, :bad_request)["errors"] == "Invalid request body structure"
+    end
+
+    test "renders error when filePath is invalid", %{
+      conn: conn,
+      room_id: room_id
+    } do
+      conn =
+        post(conn, ~p"/room/#{room_id}/component",
+          type: "file",
+          options: %{filePath: "some/fake/path.h264"}
+        )
+
+      assert json_response(conn, :bad_request)["errors"] == "Invalid request body structure"
+    end
+  end
+
```

Comment:
I suggest adding one more test for the file path that leads outside of base_path